MCprep_addon/spawner/models.py

```diff
@@ -0,0 +1,1350 @@
+					new_target.animation_data_create()
+					new_target.animation_data.action = old_target.animation_data.action
+					conf.log("Updated animation of armature for instance of "+src_coll.name)
+
+				if mod.object in new_objs:
+					continue # was already the new object target for modifier
+				mod.object = new_target
+				conf.log("Updated target of armature for instance of "+src_coll.name)
+
+	def prep_collection(self, context, block, pre_groups):
+		"""Prep the imported collection, ran only if newly imported (not cached)"""
+
+		# Group method first, move append to according layer
+		for ob in util.get_objects_conext(context):
+			util.select_set(ob, False)
+
+		layers = [False]*20
+		if not hasattr(context.scene, "layers"):
+			# TODO: here add all subcollections to an MCprepLib collection.
+			pass
+		elif self.append_layer==0:
+			layers = context.scene.layers
+		else:
+			layers[self.append_layer-1] = True
+		objlist = []
+		group = None
+		for coll in util.collections():
+			if coll in pre_groups:
+				continue
+			for ob in coll.objects:
+				if ob.name in context.scene.objects:
+					#context.scene.objects.unlink(ob)
+					objlist.append(ob)
+			if util.nameGeneralize(coll.name) == util.nameGeneralize(block):
+				group = coll
+
+		if group is None:
+			return None
+
+		if self.prep_materials is True:
+			for ob in objlist:
+				util.select_set(ob, True)
+			try:
+				bpy.ops.mcprep.prep_materials(
+					autoFindMissingTextures=False,
+					improveUiSettings=False,
+					skipUsage=True) # if cycles
+			except:
+				print("Could not prep materials")
+			for ob in util.get_objects_conext(context):
+				util.select_set(ob, False)
+
+		if hasattr(context.scene, "layers"): # 2.7 only
+			for obj in objlist:
+				obj.layers = layers
+		return group
+
+
+class MCPREP_OT_reload_models(bpy.types.Operator):
+	"""Force reload the Model objects and cache."""
+	bl_idname = "mcprep.reload_models"
+	bl_label = "Reload Models"
+
+	@tracking.report_error
+	def execute(self, context):
+		if not context.scene.model_path.lower().endswith('.blend'):
+			self.report({'WARNING'}, "Meshswap file must be a .blend, try resetting")
+		elif not os.path.isfile(bpy.path.abspath(context.scene.model_path)):
+			self.report({'WARNING'}, "Meshswap blend file does not exist")
+
+		# still reload even with above issues, cache/UI should be cleared
+		get_model_cache(context, clear=True)
+		updateModelList(context)
+		return {'FINISHED'}
+
+
+class MCPREP_UL_model_list(bpy.types.UIList):
+	"""For asset listing UIList drawing"""
+	# previously McprepMeshswap_UIList
+	def draw_item(self, context, layout, data, set, icon, active_data, active_propname, index):
+
+		if self.layout_type in {'DEFAULT', 'COMPACT'}:
+			col = layout.column()
+			col.prop(set, "name", text="", emboss=False)
+
+		elif self.layout_type in {'GRID'}:
+			layout.alignment = 'CENTER'
+			layout.label(text="", icon='QUESTION')
+
+
+class MCPREP_OT_model(bpy.types.Operator):
```

Comment:
Should this operator be entirely deleted?

As I understand, the models should only be available as something you can "spawn" from the spawner, thus this "meshswap" type of auto placing it into the scene is not useful. Let me know if I am wrong and you were intended that it does replace entities somehow included in world imports.

Reply:
I kind of wanted to implement it so that when meshswaping automatically, it would also replace things like minecarts. It would be an option in the meshswap options "Autorig entities" or something I was going to reference that class from the other meshswap file somehow. Probably not going to happen though. I should probably delete that